owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.kt

```diff
@@ -0,0 +1,451 @@
+            }
+        }
+
+        fun dismissLoadingDialog() {
+            val frag = activity?.supportFragmentManager?.findFragmentByTag(DIALOG_WAIT_TAG)
+            if (frag != null) {
+                val loading = frag as LoadingDialog
+                loading.dismiss()
+            }
+        }
+
+        private fun showPreviewText(
+            text: String,
+            mimeType: String,
+            rootView: RelativeLayout,
+            textView: TextView,
+            textLayout: View,
+            tabLayout: TabLayout,
+            viewPager: ViewPager2
+        ) {
+            if (mimeType == "text/markdown") {
+                rootView.removeView(textLayout)
+                showFormatType(text, mimeType, tabLayout, viewPager)
+                tabLayout.visibility = View.VISIBLE
+                viewPager.visibility = View.VISIBLE
+            } else {
+                rootView.removeView(tabLayout)
+                rootView.removeView(viewPager)
+                textView.text = text
+                textLayout.visibility = View.VISIBLE
+            }
+        }
+
+        private fun showFormatType(
+            text: String,
+            mimeType: String,
+            tabLayout: TabLayout,
+            viewPager: ViewPager2
+        ) {
+            val adapter = PreviewFormatTextFragmentStateAdapter(this@PreviewTextFragment, text, mimeType)
+            viewPager.adapter = adapter
+
+            TabLayoutMediator(tabLayout, viewPager) { tab: TabLayout.Tab, position: Int ->
+                if (position == 0) {
+                    tab.text = adapter.formatTypes[mimeType]
+                } else {
+                    tab.text = adapter.formatTypes[PreviewFormatTextFragmentStateAdapter.TYPE_PLAIN]
+                }
+            }.attach()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.file_actions_menu, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+
+        if (mContainerActivity.storageManager != null) {
+            val safeFile = file
+            val accountName = mContainerActivity.storageManager.account.name
+            previewTextViewModel.filterMenuOptions(safeFile, accountName)
+
+            collectLatestLifecycleFlow(previewTextViewModel.menuOptions) { menuOptions ->
+                val hasWritePermission = safeFile.hasWritePermission
+                menu.filterMenuOptions(menuOptions, hasWritePermission)
+            }
+        }
+
+        menu.findItem(R.id.action_search)?.apply {
+            isVisible = false
+            isEnabled = false
+        }
+
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.action_share_file -> {
+                mContainerActivity.fileOperationsHelper.showShareFile(file)
+                return true
```

Comment:
We can omit the `return` keyword in each of the cases of this switch

Reply:
I removed all the returns that were in switch structure

Reply:
But the super function `onOptionsItemSelected` return a `Boolean` so it should be the same in this case, right?